store/src/java/com/zimbra/cs/account/EmailChannel.java

```diff
@@ -60,15 +61,15 @@ public String getRecoveryAccount(Account account) throws ServiceException {
     public void sendAndStoreResetPasswordRecoveryCode(ZimbraSoapContext zsc, Account account,
             Map<String, String> recoveryCodeMap) throws ServiceException {
         DateTimeFormatter formatter = DateTimeFormatter.ofPattern("EEE, d MMM yyyy HH:mm:ss z")
-                .withZone(ZoneId.of("GMT"));
+                .withZone(ZoneId.of(Util.getAccountTimeZone(account).getID()));
```

Comment:
Save the result of the `withZone` argument and use below instead of recalculating.